golang/thrift/context.go

```diff
@@ -70,6 +70,15 @@ func NewContext(timeout time.Duration) (Context, context.CancelFunc) {
 	}, cancel
 }
 
+// WrapContextForTest returns a Context that is associated with the call.
+// This should be used in units test only.
+func WrapContextForTest(ctx context.Context, call tchannel.IncomingCall) Context {
```

Comment:
optional: instead of exposing a test-specific method here, what about if we exposed
Wrap(ctx context.Context) Context

And then the caller would create the context using tchannel.WrapContextForTest, and then the tchannel/json implementations of context don't need to worry about exposing these interfaces.